src/io/flutter/inspector/DiagnosticsNode.java

```diff
@@ -358,6 +358,17 @@ boolean hasException() {
     return json.has("exception");
   }
 
+  public boolean hasCreationLocation() {
+    return json.has("creationLocation");
+  }
+
+  public Location getCreationLocation() {
+    if (!hasCreationLocation()) {
+      return null;
+    }
+    return new Location(json.getAsJsonObject("creationLocation"), null);
```

Comment:
Not sure how frequently this is called; consider caching the `Location` instance in a private field?

Reply:
Went ahead and cached the Location object.